src/parse.rs

```diff
@@ -321,6 +321,69 @@ pub fn parse_mailbox(
     }
 }
 
+pub fn parse_status(
+    mut lines: &[u8],
+    mailbox_name: &str,
+    unsolicited: &mut mpsc::Sender<UnsolicitedResponse>,
+) -> Result<Mailbox> {
+    let mut mailbox_opt = None;
+
+    loop {
+        if lines.is_empty() {
+            break mailbox_opt.ok_or_else(|| {
```

Comment:
Can this just be `while !lines.is_empty()` with the `mailbox_opt.ok_or_else` placed below the loop?